### ISSUE TITLE
Add "supportsSound" to package.json

### ISSUE THREAD

**Issue by jbphet** [phetsims/chipper#806]
Currently, a sim is configured to support sound either through an option in the `Sim.js` constructor called `supportsSound` or through a query parameter of the same name.  This approach is causing some problems during loading and initialization where we want to avoid processing and loading sounds if the sim doesn't support it, since the global that is set by `Sim.js` - i.e. `phet.joist.sim.supportsSound` - isn't available until the `Sim` object instance has been constructed.  It would make it much easier to skip loading and decoding of sounds if it could be known at load time whether sound is supported.  One way to do this would be to add a `supportsSound` configuration item to `package.json`.

There is a related parameter called `supportsEnhancedSound` that is primarily used to decide whether to add the `Enhanced Sound` item to the options menu.  This could be left as an option to the `Sim.js` constructor.

**Comment by jbphet** [comment]
Marking for developer meeting to get input as to whether this is doable and seems like a good approach.

**Comment by zepumph** [comment]
This seems reasonable to me.

**Comment by pixelzoom** [comment]
> ... to get input as to whether this is doable and seems like a good approach.

Seems doable and not bad.  Is this what's already being done for a11y with the `accessible` key in package.json? Can that be used as a model?  Are there alternatives?

> There is a related parameter called `supportsEnhancedSound` that is primarily used to decide whether to add the Enhanced Sound item to the options menu. This could be left as an option to the Sim.js constructor.

If we adopt this approach for `supportsSound`,  why not do the same for `supportsEnhancedSound`, at the same time? Seems like a win to handle them uniformly.

**Comment by zepumph** [comment]
Currently the `accessible` key is only responsible for keeping a list of all sims we want to have an `a11y-view` html for.

**Comment by jbphet** [comment]
This was discussed at the 10/2/2018 developer meeting, here is what we concluded:

+ @jpphet should go ahead and add `supportsSound` to `package.json` and remove it as an option to `Sim.js`
+ We should also put `supportsEnhancedSound` in `package.json` for consistency.  There isn't a concern about cluttering up `package.json`, so if something is an attribute of a sim, it is fine to put it in the package file, and these parameters meet that criterion
+ At least for now, we will always include the sounds in the built version of the sim and still support the query parameter for turning on the sound when it isn't on by default.  This allows us to publish dev versions where sound designs can be evaluated.  If we later decide that the sounds are adding too much additional size to the built files, we can change this.  If that occurs, two possible approaches were discussed to still allow us to create dev versions where sound could be enabled:
  + If `supportsSound` is set to false, sounds aren't included in the build, but if it is undefined, sounds **would** be built in and the query param would work
  + We could build the sounds into the `_all` file even if `supportsSound` wasn't set to true, and this file could be used to evaluate prototype sound designs
+ We discussed whether to nest the options with accessibility options or under their own parent field, such as `sound`.  Sound isn't purely about accessibility, so it doesn't seem appropriate to make it part of an accessibility config option.  Ultimately, it seemed simpler to keep them as two separate top-level sim configuration options, so that's what we should do for now.
